### PR TITLE
Add standard test related scripts to build repo

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add standard test related scripts to build repo.
This PR include all the related integTest scripts for deploy cluster, setup mavenlocal, as well as integTest dockerfile.
These are all the things we need to run integTest in opensearch/dashboards.

 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tools/standard-test/integtest-runner.al2.dockerfile [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+USER 1000
+WORKDIR /usr/share/opensearch
+
+# Hard code node version and yarn version for now
+# nvm environment variables
+ENV NVM_DIR /usr/share/opensearch/.nvm
+ENV NODE_VERSION 10.23.1
+# install nvm
+# https://github.com/creationix/nvm#install-script
+RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.31.2/install.sh | bash
```
Comment (camerski) [inline tools/standard-test/integtest-runner.al2.dockerfile:80]: Is there really no better way to install nvm than to download a random script from the internet and run it as root?

File: tools/standard-test/integtest-runner.al2.dockerfile [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+RUN yum install -y xorg-x11-server-Xvfb gtk2-devel gtk3-devel libnotify-devel GConf2 nss libXScrnSaver alsa-lib
+
+# Add Notebook dependencies
+RUN yum install -y libnss3.so xorg-x11-fonts-100dpi xorg-x11-fonts-75dpi xorg-x11-utils xorg-x11-fonts-cyrillic xorg-x11-fonts-Type1 xorg-x11-fonts-misc fontconfig freetype && yum clean all
+
+# Add Yarn dependencies
+RUN yum groupinstall -y "Development Tools"
+
+# Remove yum cache
+RUN yum clean all && rm -rf /var/cache/yum/*
```
Comment (camerski) [inline tools/standard-test/integtest-runner.al2.dockerfile:50]: Why is this necessary?

File: tools/standard-test/integtest-runner.al2.dockerfile [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+# Setup Shared Memory
+RUN  chmod -R 777 /dev/shm
+
+# Must have values for ulimits on the server
+# sudo sysctl -w vm.max_map_count=262144
+# ulimit -n 65535
+# Docker /etc/sysconfig/docker has to have nofiles to 65535
+# OPTIONS="--default-ulimit nofile=65535:65535"
```
Comment (camerski) [inline tools/standard-test/integtest-runner.al2.dockerfile:59]: Why is this all commented out?

File: tools/standard-test/integtest-runner.al2.dockerfile [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+# Create user group
+RUN groupadd -g 1000 opensearch && \
+    useradd -u 1000 -g 1000 -d /usr/share/opensearch opensearch && \
+    mkdir -p /usr/share/opensearch && \
+    chown -R 1000:1000 /usr/share/opensearch
+
+# Environment variables JAVA
+RUN echo "export JAVA_HOME=`dirname $(dirname $(readlink -f $(which javac)))`" >> /etc/profile.d/java_home.sh
```
Comment (camerski) [inline tools/standard-test/integtest-runner.al2.dockerfile:68]: Doesn't installing AdoptOpenJDK set up JAVA_HOME automatically?

File: lib/shell/common.sh [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+      else
+          echo -e "\tProcess $pid_kill not exist"
+      fi
+
+    done
+}
+
+function Temp_Folder_Create() {
+    TEMP_SUFFIX="_TEMP"
+    TEMP_FOLDER=$(mktemp --suffix=$TEMP_SUFFIX -d)
```
Comment (camerski) [inline lib/shell/common.sh:76]: TEMP_SUFFIX is unnecessary. Just let `mktemp` do its thing.

File: lib/shell/common.sh [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+function Trap_File_Delete() {
+    echo "Trap deletion of $TEMP_FOLDER for these signals: $TRAP_SIG_LIST"
+    trap '{ echo Attempt to rm "$TEMP_FOLDER"; if [ ! -z "$TEMP_FOLDER" ]; then echo Removing "$TEMP_FOLDER"; rm -rf -- "$TEMP_FOLDER"; else echo No File/Folder Exist; fi;}' $TRAP_SIG_LIST
+    
+}
+
+function Trap_File_Delete_No_Sigchld() {
+    echo "Trap deletion of $TEMP_FOLDER for these signals: $TRAP_SIG_LIST_NO_SIGCHLD"
+    trap '{ echo Attempt to rm "$TEMP_FOLDER"; if [ ! -z "$TEMP_FOLDER" ]; then echo Removing "$TEMP_FOLDER"; rm -rf -- "$TEMP_FOLDER"; else echo No File/Folder Exist; fi;}' $TRAP_SIG_LIST_NO_SIGCHLD
```
Comment (camerski) [inline lib/shell/common.sh:89]: Re-use the Delete_Temp_Folder function

File: lib/shell/common.sh [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+function Temp_Folder_Create() {
+    TEMP_SUFFIX="_TEMP"
+    TEMP_FOLDER=$(mktemp --suffix=$TEMP_SUFFIX -d)
+    echo $TEMP_FOLDER
+    
+}
+
+function Trap_File_Delete() {
+    echo "Trap deletion of $TEMP_FOLDER for these signals: $TRAP_SIG_LIST"
+    trap '{ echo Attempt to rm "$TEMP_FOLDER"; if [ ! -z "$TEMP_FOLDER" ]; then echo Removing "$TEMP_FOLDER"; rm -rf -- "$TEMP_FOLDER"; else echo No File/Folder Exist; fi;}' $TRAP_SIG_LIST
```
Comment (camerski) [inline lib/shell/common.sh:83]: That's a lot of code in one line...I recommend pulling it into a function:

```
function Delete_Temp_Folder {
  echo Attempt to rm "$TEMP_FOLDER"
  ...
}

trap Delete_Temp_Folder $TRAP_SIG_LIST
```

File: lib/shell/common.sh [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+
+function Trap_Wait_Term_Cleanup() {
+    # Reset -e so that the process does not quit in the middle of cleanup
+    set +e
+    echo "PID List: ${PARENT_PID_LIST[@]}"
+    echo "Working Directory: $@"
+    echo "Trap and Wait for these signals: ${TRAP_SIG_LIST}"
+
+    trap '{ Kill_Process_PID ${PARENT_PID_LIST[@]}; echo Removing $TEMP_FOLDER; rm -rf -- "$TEMP_FOLDER"; }' $TRAP_SIG_LIST
```
Comment (camerski) [inline lib/shell/common.sh:109]: Re-use the Delete_Temp_Folder function

File: lib/shell/common.sh [hunk 9]
```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
```
Comment (camerski) [inline lib/shell/common.sh:1]: Rename from `common.sh` to something more descriptive...maybe `process_control.sh` or something like that.

`common.sh` doesn't tell me what this code does.

File: tools/standard-test/integtest-runner.al2.dockerfile [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+# permissions and limitations under the License.
+
+# This is a docker image specifically for standardize the test environment
+# for both developers and ci/cd tools in OpenSearch
+
+
+FROM amazonlinux:2
+
+# Replace shell with bash so we can source files
+RUN rm /bin/sh && ln -s /bin/bash /bin/sh
```
Comment (camerski) [inline tools/standard-test/integtest-runner.al2.dockerfile:31]: Why do we need this? The shebangs all specify `/bin/bash`, so why do we need to replace `sh`?

File: tools/standard-test/integtest_dependencies_opensearch.sh [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+# https://github.com/opensearch-project/OpenSearch
+# $1 is the local repo that needs deployment
+# $2 is the version number such as 1.0.0, 2.0.0
+# $3 is the version qualifier such as beta1, rc1
+
+if [ -z "$1" ] || [ -z "$2" ]
+then
+  echo "You must specify parameters 'ACTION VERSION [QUALIFIER]'"
+  echo "Example: $0 os 1.0.0"
+  echo "Example: $0 js 1.0.0 rc1"
```
Comment (camerski) [inline tools/standard-test/integtest_dependencies_opensearch.sh:37]: List the possible actions and explain what they mean.

File: lib/shell/common.sh [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
```
Comment (gaiksaya) [inline lib/shell/common.sh:24]: Not required. Can you remove this from all files? Makes its difficult to read the code

File: tools/standard-test/integtest-runner.al2.dockerfile (hunk 10 above)
Comment (peterzhuamazon) [reply tools/standard-test/integtest-runner.al2.dockerfile:31]: Remove it.

File: tools/standard-test/integtest-runner.al2.dockerfile (hunk 2 above)
Comment (peterzhuamazon) [reply tools/standard-test/integtest-runner.al2.dockerfile:50]: Improved comments.

File: tools/standard-test/integtest-runner.al2.dockerfile (hunk 3 above)
Comment (peterzhuamazon) [reply tools/standard-test/integtest-runner.al2.dockerfile:59]: Moved to the top as a comment to LINUX host user.

File: tools/standard-test/integtest-runner.al2.dockerfile (hunk 4 above)
Comment (peterzhuamazon) [reply tools/standard-test/integtest-runner.al2.dockerfile:68]: Nope.

File: tools/standard-test/integtest-runner.al2.dockerfile (hunk 1 above)
Comment (peternied) [reply tools/standard-test/integtest-runner.al2.dockerfile:83]: At the very least we could put this script into this repository so we know it isn't changing

File: tools/standard-test/setup_limits.sh [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+# This is necessary for servers that try to run OpenSearch with docker
+# The hosts have to have these settings ready
+# Or the OpenSearch Process will not run
+sudo sysctl -w vm.max_map_count=262144
+ulimit -n 65535
+# Required for PA and similar processes that require shared memory to work
+sudo chmod -R 777 /dev/shm
+
+# As for docker the host must have these as well in /etc/sysconfig/docker and restart service
+# OPTIONS="--default-ulimit nofile=65535:65535"
```
Comment (peternied) [inline tools/standard-test/setup_limits.sh:35]: Not required? Can we remove

File: tools/standard-test/opensearch-onetime-setup.sh [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+echo $OPENSEARCH_HOME
+
+##Security Plugin
+SECURITY_PLUGIN="opensearch-security"
+bash $OPENSEARCH_HOME/plugins/$SECURITY_PLUGIN/tools/install_demo_configuration.sh -y -i -s
+
+##Perf Plugin
+PA_PLUGIN="opensearch-performance-analyzer"
+chmod 755 $OPENSEARCH_HOME/plugins/$PA_PLUGIN/pa_bin/performance-analyzer-agent
+chmod -R 755 /dev/shm
```
Comment (peternied) [inline tools/standard-test/opensearch-onetime-setup.sh:44]: Is this duplicated with the same setting in `tools/standard-test/setup_limits.sh`?

**Comment by peterzhuamazon** [comment]
@peternied I have another revision coming up after talking to @camerski this morning.
All the things you are currently reviewing is subject to change.
Please wait until the revision come up. Thanks.

File: tools/standard-test/opensearch-onetime-setup.sh [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+
+# This script performs one-time setup for the OpenSearch tarball distribution.
+# It installs a demo security config and sets up the performance analyzer
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply tools/standard-test/opensearch-onetime-setup.sh:28]: Will retrieve from scripts folder now.

File: tools/standard-test/integtest-runner.al2.dockerfile [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+    && nvm use default
+# add node and npm to path so the commands are available
+ENV NODE_PATH $NVM_DIR/v$NODE_VERSION/lib/node_modules
+ENV PATH $NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH
+# install yarn
+RUN npm install -g yarn@^1.22.1
+# confirm installation
+RUN node -v
+RUN npm -v
+RUN yarn -v
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply tools/standard-test/integtest-runner.al2.dockerfile]: Explained with @camerski and we have agreed this is the best way for a docker build to fail if any part of the installation has issues.

File: tools/standard-test/integtest-runner.al2.dockerfile (hunk 1 above)
Comment (peterzhuamazon) [reply tools/standard-test/integtest-runner.al2.dockerfile:83]: This is the best way to install this.
I will save this script to local for now.

Comment (peterzhuamazon) [reply tools/standard-test/integtest-runner.al2.dockerfile:83]:
```
 => [12/19] RUN bash nvm_install.sh                                                                                                                                                                                                                                                                                    3.4s
```

File: lib/shell/common.sh (hunk 5 above)
Comment (peterzhuamazon) [reply lib/shell/common.sh]: This is used in the old cleanup scripts to remove everything have that suffix.
Will remove.

File: lib/shell/common.sh (hunk 6 above)
Comment (peterzhuamazon) [reply lib/shell/common.sh]: In the new design this would require process_control.sh to source folder_management.sh.
Will keep this as of now.

File: lib/shell/common.sh (hunk 7 above)
Comment (peterzhuamazon) [reply lib/shell/common.sh]: Thanks, change codes.

File: lib/shell/common.sh (hunk 8 above)
Comment (peterzhuamazon) [reply lib/shell/common.sh]: In the new design this would require process_control.sh to source folder_management.sh.
Will keep this as of now.

File: lib/shell/common.sh (hunk 9 above)
Comment (peterzhuamazon) [reply lib/shell/common.sh]: Renamed already. Thanks.

File: tools/standard-test/integtest_dependencies_opensearch.sh (hunk 11 above)
Comment (peterzhuamazon) [reply tools/standard-test/integtest_dependencies_opensearch.sh]: Listed with more details already in code.

File: lib/shell/common.sh (hunk 12 above)
Comment (peterzhuamazon) [reply lib/shell/common.sh]: Thanks.

File: tools/standard-test/setup_limits.sh [hunk 17]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (peternied) [inline tools/standard-test/setup_limits.sh:1]: How is this script used, seems like the docker file implies some of these settings should be updated

File: tools/standard-test/setup_limits.sh (hunk 13 above)
Comment (peterzhuamazon) [reply tools/standard-test/setup_limits.sh:15]: This is the requirements for LINUX host, I have already move these comments to the top of the file.

File: tools/standard-test/opensearch-onetime-setup.sh (hunk 14 above)
Comment (peterzhuamazon) [reply tools/standard-test/opensearch-onetime-setup.sh]: Already remove file.

File: tools/standard-test/setup_limits.sh (hunk 17 above)
Comment (peterzhuamazon) [reply tools/standard-test/setup_limits.sh:1]: Will move to the README file after talking to you.

**Review by peternied — APPROVED** [review]

**Review by bbarani — APPROVED** [review]
